Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -357,6 +357,11 @@ public struct ConvertAction: Action, RecreatingContext {
     /// Converts each eligible file from the source documentation bundle,
     /// saves the results in the given output alongside the template files.
     mutating public func perform(logHandle: LogHandle) throws -> ActionResult {
+        defer {
+            diagnosticEngine.finalize()
```

Comment:
Since this is the second time in the convert action that `finalize` is called on the diagnostic engine; I wonder if we should rename is to `flush` instead (and the same with `DiagnosticConsumer`) to indicate that it's expected to be called more than once.

Reply:
Note; since `finalize()` is public API we'd need to keep it but mark it deprecated. It would be a breaking change to remove it.

Reply:
Do we just want to remove the previous call to `finalize`? Seems weird that we finalize the diagnostic engine before building up the index, do coverage, etc?

Reply:
I think that it would be much cleaner if we emit all the diagnostics inside the convert action instead of instantiating two different diagnostic engines for the same process. This might require a little bit more refactoring, but is there any reason not to do this?  @d-ronnqvist 

Renaming to `flush` is also ok, but I'm leaning more toward the other implementation.

Reply:
Renamed to `flush`in https://github.com/apple/swift-docc/pull/664/commits/42143d2a787dc347376e811a6b1c324e82817285